runtime/compiler/env/VMJ9Server.cpp

```diff
@@ -400,9 +400,24 @@ TR_J9ServerVM::getComponentClassFromArrayClass(TR_OpaqueClassBlock *arrayClass)
 bool
 TR_J9ServerVM::classHasBeenReplaced(TR_OpaqueClassBlock *clazz)
    {
-   JITaaS::J9ServerStream *stream = _compInfoPT->getMethodBeingCompiled()->_stream;
-   stream->write(JITaaS::J9ServerMessageType::VM_classHasBeenReplaced, clazz);
-   return std::get<0>(stream->read<bool>());
+   bool classReplacedFromCache = false;
+
```

Comment:
You need to acquire this monitor when accessing the ROMClassMap
`OMR::CriticalSection getRemoteROMClass(clientSessionData->getROMMapMonitor());`